templates/hooks/workgroup/sdk_read_one_post_request.go.tpl

```diff
@@ -0,0 +1,7 @@
+	// Athena API returns a 400 instead of a 404 when a workgroup is not found.
+	if err != nil {
+		if reqErr, ok := ackerr.AWSRequestFailure(err); ok && reqErr.StatusCode() == 400 && 
+			strings.Contains(reqErr.Error(), "is not found") {
+			return nil, ackerr.NotFound
+		}
+	}
```

Comment:
We have a code generator option that helps remapping the code errors https://aws-controllers-k8s.github.io/community/docs/contributor-docs/code-generator-config/#exceptions-correcting-exception-codes